### PR TITLE
Removing UnicodeCodePoint and Utf8CodeUnit classes

### PR DESCRIPTION
**Pull request by shiftylogic**
* Unicode code points are represented as uint values for now.
* UTF8 logic is isolated to where it was needed.
* Minor other cosmetic changes to facilitate this change and cleanup.

### CODE REVIEW

**Comment by shiftylogic** [comment]
@ahsonkhan @KrzysztofCwalina

File: tests/System.Text.Primitives.Tests/Encoding/PerformanceTests.cs [hunk 1]
```diff
@@ -251,7 +251,7 @@ public unsafe void ReverseEnumerateCodePointsConstructFromSpan()
                         Utf8String.CodePointReverseEnumerator it = utf8s.CodePoints.GetReverseEnumerator();
                         while (it.MoveNext())
                         {
-                            UnicodeCodePoint codePoint = it.Current;
+                            var codePoint = it.Current;
```
Comment (KrzysztofCwalina) [inline tests/System.Text.Primitives.Tests/Encoding/PerformanceTests.cs]: We don't use var unless it's super clear what the type is. Here it does not seem to be super clear.

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ahsonkhan** [comment]
> We don't use var unless it's super clear what the type is. Here it does not seem to be super clear.
@KrzysztofCwalina, we merged this without addressing your comment?

**Comment by shiftylogic** [comment]
@ahsonkhan I've fixed the use of 'var' in my own branch. It will come in with my next PR tomorrow.

**Pull request merged**